.github/workflows/rust.yml

```diff
@@ -11,57 +11,57 @@ on:
 
 jobs:
   build:
-    name: Build
+    name: Tests & Build
     env:
+      # `-D warnings` means any warnings emitted will cause build to fail
       RUSTFLAGS: "-D warnings -C opt-level=z -C target-cpu=x86-64 -C debuginfo=1"
       X86_64_PC_WINDOWS_MSVC_OPENSSL_DIR: c:/vcpkg/installed/x64-windows
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        # was commented bc of too much spend on Macos which is counted 10x, Windows is 2x
         os: [macos-latest, windows-latest, ubuntu-latest]
 
     steps:
       - name: Checkout
         uses: actions/checkout@v1
 
-      - name: Install Last Stable Rust
+      - name: Install last stable Rust
         uses: actions-rs/toolchain@v1
         with:
           toolchain: stable
           components: rustfmt, clippy
 
-      - name: cargo tree - to check lockfile validity
+      - name: Check lockfile
         uses: actions-rs/cargo@v1
         with:
           command: tree
           args: --locked
 
-      - name: cargo fmt
+      - name: Check formatting
         uses: actions-rs/cargo@v1
         with:
           command: fmt
           args: --all -- --check
 
-      - name: Install openssl
-        if: matrix.os == 'windows-latest'
+      - name: Install openssl ( Windows only )
+        if: runner.os == 'Windows'
         run: |
           vcpkg install openssl:x64-windows openssl:x64-windows-static
           vcpkg list
           vcpkg integrate install
 
-      - name: cargo test
+      - name: Unit tests
         uses: actions-rs/cargo@v1
         with:
           command: test
           args: --workspace --locked
 
-      - name: cargo test sgx
-        if: matrix.os == 'ubuntu-latest'
+      - name: Unit tests for SGX
+        if: ${{ runner.os == 'Linux' && startsWith( github.head_ref, 'sgx/' ) }}
```

Comment:
I've disabled it since it is an optional feature.
I think we can introduce a separate workflow and make it not required.
What do you think?

Reply:
TLDR: I'm against making these tests optional

This feature is optional because of cargo semantics - we have 2 kinds of builds, where the second one is Graphene / SGX exclusive. There are some alterations and limitations to the computation flow which would cripple the feature-set of the former and had to be implemented differently or disabled in the latter.

There aren't many SGX-specific tests, mainly due to hardware requirements of a test environment. In general, we run the same test suite to prevent regressions in the alternative flow and make sure that the application still builds with the `sgx` feature on.

Making the test suite optional means that we allow developers to break this feature.